### ISSUE TITLE
Schema documentation

### ISSUE THREAD

**Issue by nokome** [stencila/schema#26]
Some options for generating HTML documentation:

1.  Generate docs from JSON Schema directly
  - [@cloudflare/doca](https://github.com/cloudflare/json-schema-tools#applications-and-application-support) which appears to be used for https://api.cloudflare.com. See https://blog.cloudflare.com/cloudflares-json-powered-documentation-generator/.
  - The https://frictionlessdata.io/specs/ appear to be generated from the schemas in https://github.com/frictionlessdata/specs (home-grown doc generator?)
  - https://github.com/lbovet/docson

2. Generate docs from an OpenAPI spec
  - JSON Schema is used with the https://github.com/OAI/OpenAPI-Specification and there are several good documentation tools around it
  - We could create a simple REST API specification with say `GET` endpoints for each type.
  - https://github.com/Rebilly/ReDoc#readme. We have used ReDoc and this approach in [this repo previously](https://github.com/stencila/schema/blob/099498f6f160138df586e1ba37e52fa9f6400a28/gulpfile.js)
  - https://github.com/Mermade/shins e.g. https://api-reference.shutterstock.com/
  - https://sourcey.com/spectacle/
  - https://swagger.io/solutions/api-documentation/

3. Generate docs from Typescript definitions
  - We could use https://typedoc.org/ to generate docs from the generated Typescript docs, but this adds another level of indeirection and should probably be avoided.

**Comment by jwijay** [comment]
4. Roll our own solution.

**Comment by alex-ketch** [comment]
As part of the design refinement process we should explore removing the focus from the reference JSON schema definition, and focus on a narrative/text based layout instead.
In this variation we might want to put the schema JSON references at the bottom of the page, or simply link to them.

**Comment by nokome** [comment]
Over the last few months of working on these schema definitions I've been vacillating between alternatives for documenting them. This has also been in the context of also considering how best to develop and maintain documentation for our other projects e.g [`encoda`](https://github.com/stencila/encoda) and the higher level documentation of the https://stenci.la website.

Of the above options, we went with (4) Roll our own solution (using Gatsby). We have expressed a desire to have less technical, more narrative-style docs https://github.com/stencila/schema/issues/45. But we've struggled to find the time to write those docs - which, by their nature, take more time than auto generated docs. We've also had issues with making sure that examples in docs are valid.

More recently, I've experimented with a dog-food approach, writing documentation in Markdown and then generating publishable HTML using Encoda. This has the advantage of being able to make use of Encoda's functions for inclusion and validation - as well as checking if our dog food tastes OK!

I am now leaning towards an approach of (a) auto-generated, API-style, developer-focused docs from source code in library repos e.g. `schema`, `encoda`, `dockta` using TypeDoc etc (b) hand written, higher level, guide-style, user-focused docs in `stencila`.

For this repo, that would mean, instead of having `.md` documents, we would put extended `description`s and `example` in the `.schema.yaml` files and then generate docs directly from there. I propose that we would take a dog-food approach here too, and instead of using something like `doca` or `docson`, add a task in our `Gulpfile.js` that takes the YAML/JSON, transforms it into a `Article` and publishes it as HTML using `encoda` and our default theme, or perhaps a `stencila-docs` theme. Other higher level docs, specific to this repo would go into the README.md and get published as an index page.

I see a need for at least simple cross-page (== files in a folder == breadcrumbs) and within-page (== headings in a file == TOC) navigation for user projects (e.g. `stencila publish ./myproject` generates a navigable tree of HTML pages for that project) much like https://bookdown.org/yihui/bookdown/usage.html (?) and other static site generators do. So, it seems like a better approach to invest in implementing that in Stencila and then using it as dog-food in this repo, rather than maintaining Gatsby generated docs, which may have better UX/UI but which will take some effort to develop and maintain.
